### PR TITLE
OPRUN-3606: Add diagnostics to olmv1 tests

### PR DESCRIPTION
**Pull request by tmshort**
* Add diagnostics to figure out what happened when context timeout occurs
* Remove [Serial] requirements, the install tests can now happen in parallel

### CODE REVIEW

File: test/extended/olm/olmv1.go [hunk 1]
```diff
@@ -150,200 +159,220 @@ var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Serial][Skipped:Disconnec
...
+		return false, fmt.Errorf("Progressing condtion should not be nil"), ""
 	}
 	if c.Reason != "Retrying" {
-		return false, nil
+		return false, nil, fmt.Sprintf("reason not %q: %q", "Retrying", c.Reason)
 	}
 	if !strings.Contains(c.Message, "error creating image source") {
-		return false, nil
+		return false, nil, fmt.Sprintf("message does not contain %q: %q", "error creating image source", c.Message)
 	}
```
[the remark this answers is not in the document]
Comment (everettraven) [reply test/extended/olm/olmv1.go:374]: This applies to all the `waitFor*` functions

File: test/extended/olm/olmv1.go [hunk 2]
```diff
@@ -150,200 +159,220 @@ var _ = g.Describe("[sig-olmv1][OCPFeatureGate:NewOLM][Serial][Skipped:Disconnec
...
-	err = json.Unmarshal([]byte(output), &conditions)
-	if err != nil {
-		return false, fmt.Errorf("error in json.Unmarshal(%v): %v", output, err)
+	if err := json.Unmarshal([]byte(output), &conditions); err != nil {
+		return false, fmt.Errorf("error in json.Unmarshal(%v): %v", output, err), ""
 	}
 	if !meta.IsStatusConditionPresentAndEqual(conditions, "Progressing", metav1.ConditionTrue) {
-		return false, nil
+		return false, nil, "progressing not present or not true"
 	}
```
[the remark this answers is not in the document]
Comment (tmshort) [reply test/extended/olm/olmv1.go:364]: It's a convenience function provided by `meta`, but I see your point.

File: test/extended/olm/olmv1.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmshort) [reply test/extended/olm/olmv1.go:374]: Fair point.

**Review by everettraven — APPROVED** [review]
/lgtm

**Comment by grokspawn** [comment]
/retitle OPRUN-3606: Add diagnostics to olmv1 tests

**Pull request merged**